src/dodal/devices/oav/pin_tip_detection/oav_with_pin_tip_detection.py

```diff
@@ -0,0 +1,208 @@
+    ArrayProcessingFunctions,
+    MxSampleDetect,
+)
+from dodal.log import LOGGER
+
+T = TypeVar("T")
+
+
+def _create_soft_signal(
+    name: str, datatype: T | Type[T], initial_value: T
+) -> SignalRW[T]:
+    class _Converter(SimConverter):
+        def make_initial_value(self, *args, **kwargs) -> T:
+            return initial_value
+
+    class _SimSignalBackend(SimSignalBackend):
+        async def connect(self) -> None:
+            self.converter = _Converter()
+            self._initial_value = self.converter.make_initial_value()
+            self._severity = 0
+
+            await self.put(None)
+
+    backend = _SimSignalBackend(
+        datatype, f"sim://oav_with_pin_tip_detection_soft_params:{name}"
+    )
+    signal: SignalRW[T] = SignalRW(backend)
+    return signal
+
+
+class PinTipDetection(Readable, Device):
+    def __init__(self, prefix: str, name: str = ""):
+        self._prefix: str = prefix
+        self._name = name
+
+        self.array_data: SignalR[NDArray[np.uint8]] = epics_signal_r(
+            NDArray[np.uint8], f"pva://{prefix}PVA:ARRAY"
+        )
+
+        self.oav_width: SignalR[int] = epics_signal_r(
+            int, f"{prefix}PVA:ArraySize1_RBV"
+        )
+        self.oav_height: SignalR[int] = epics_signal_r(
+            int, f"{prefix}PVA:ArraySize2_RBV"
+        )
+
+        # Soft parameters for pin-tip detection.
+        self.timeout: SignalRW[float] = _create_soft_signal("timeout", float, 10.0)
+        self.preprocess: SignalRW[
+            Callable[[np.ndarray], np.ndarray]
+        ] = _create_soft_signal(
+            "preprocess",
+            Callable[[np.ndarray], np.ndarray],
+            ArrayProcessingFunctions.identity(),
+        )
+        self.canny_upper: SignalRW[int] = _create_soft_signal("canny_upper", int, 100)
+        self.canny_lower: SignalRW[int] = _create_soft_signal("canny_lower", int, 50)
+        self.close_ksize: SignalRW[int] = _create_soft_signal("close_ksize", int, 5)
+        self.close_iterations: SignalRW[int] = _create_soft_signal(
+            "close_iterations", int, 5
+        )
+        self.scan_direction: SignalRW[int] = _create_soft_signal(
+            "scan_direction", int, 1
+        )
+        self.min_tip_height: SignalRW[int] = _create_soft_signal(
+            "min_tip_height", int, 5
+        )
+
+        super().__init__(name=name)
+
+    async def connect(self, sim: bool = False):
+        for signal in [self.array_data, self.oav_width, self.oav_height]:
+            await signal.connect(sim=sim)
+
+        for soft_signal in [
+            self.timeout,
+            self.preprocess,
+            self.canny_upper,
+            self.canny_lower,
+            self.close_ksize,
+            self.close_iterations,
+            self.scan_direction,
+            self.min_tip_height,
+        ]:
+            await soft_signal.connect(sim=False)  # Soft signals don't need simulating.
```

Comment:
This was because passing `sim=True` into a simulated backend seemed (I think) to actually create a *new* simulated backend, undoing the custom one I'd created above with initial values etc.

I'll try refactoring into something a bit nicer...